src/main/java/de/tum/in/www1/artemis/web/rest/CourseResource.java

```diff
@@ -659,6 +665,35 @@ public void fetchParticipationsWithSubmissionsAndResultsForCourses(List<Course>
         return ResponseUtil.wrapOrNotFound(Optional.of(course));
     }
 
+    /**
+     * GET /courses/:courseId/lockedSubmissions Get locked submissions for course for user
+     *
+     * @param courseId the id of the course
+     * @return the ResponseEntity with status 200 (OK) and with body the course, or with status 404 (Not Found)
+     * @throws AccessForbiddenException if the current user doesn't have the permission to access the course
+     */
+    @GetMapping("/courses/{courseId}/lockedSubmissions")
+    @PreAuthorize("hasAnyRole('TA', 'INSTRUCTOR', 'ADMIN')")
+    public ResponseEntity<List<Submission>> getLockedSubmissionsForCourse(@PathVariable Long courseId) throws AccessForbiddenException {
+        log.debug("REST request to get all locked submissions for course : {}", courseId);
+        long start = System.currentTimeMillis();
+        Course course = courseService.findOneWithExercises(courseId);
+        User user = userService.getUserWithGroupsAndAuthorities();
+        if (!authCheckService.isAtLeastTeachingAssistantInCourse(course, user)) {
+            throw new AccessForbiddenException("You are not allowed to access this resource");
+        }
+
+        List<Submission> submissions = submissionService.getLockedSubmissions(courseId);
+
+        for (Submission submission : submissions) {
+            submissionService.hideDetails(submission, user);
+        }
+
+        long end = System.currentTimeMillis();
+        log.info("Finished /courses/" + courseId + "/submissions call in " + (end - start) + "ms");
```

Comment:
```suggestion
        log.debug("Finished /courses/" + courseId + "/submissions call in " + (end - start) + "ms");
```

Reply:
we don't really need info here. The time is tracked anyway in the metrics

Reply:
Changed it accordingly.